examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,263 @@
+"""
+============================================================
+Encoding models for visual stimuli from Miyawaki et al. 2008
+============================================================
+
+This example partly reproduces the encoding model presented in
+    `Visual image reconstruction from human brain activity
+    using a combination of multiscale local image decoders
+    <http://www.cell.com/neuron/abstract/S0896-6273%2808%2900958-6>`_,
+    Miyawaki, Y., Uchida, H., Yamashita, O., Sato, M. A.,
+    Morito, Y., Tanabe, H. C., ... & Kamitani, Y. (2008).
+    Neuron, 60(5), 915-929.
+
+Encoding models try to predict neuronal activity using a presented stimulus,
+like an image or sound. Where decoding goes from brain data to
+real-world stimulus, encoding goes the other direction.
+
+We demonstrate how to build such an **encoding model** in nilearn, predicting
+**fMRI data** from **visual stimuli**, using the dataset from `Miyawaki et al.,
+2008  <http://www.cell.com/neuron/abstract/S0896-6273%2808%2900958-6>`_.
+
+In a part of this experiment, participants were shown images consisting of
+10x10 binary (either black or white) pixels and the corresponding fMRI
+activity was recorded. We will try to predict the activity in each voxel
+from the binary pixel-values of the presented images. Then we extract the
+receptive fields for a set of voxels to see which pixel location a voxel
+is most sensitive to.
+
+See also :doc:`plot_miyawaki_reconstruction` for a decoding
+approach for the same dataset.
+"""
+
+##############################################################################
+# Loading the data
+# ----------------
+# Some general imports:
+
+import numpy as np
+import pylab as plt
```

Comment:
In the examples, we tend to import modules in the cell where we need them, in order to have the imports as close as possible to the code that uses them.